### ISSUE TITLE
Allow to create multi-cursor with keyboard

### ISSUE THREAD

**Issue by Tekbr** [microsoft/vscode#20742]
- VSCode Version: 1.10.0 - insider
- OS Version: Windows

-- Sorry for the English, I used Google Translator --

Allow me to place multiple cursors by navigating with the arrow keys.

Like this extension [multi-cursor-plus](https://atom.io/packages/multi-cursor-plus) of Atom. Press `alt + x` to highlight a cursor, use the arrow keys to scroll to the next location you want (by pressing the `alt `key), and press again.

If I wanted to take some specific cursor, I'll enter the arrows to the location and press `alt + x` again.

![68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f6b616e6b61726973746f2f61746f6d2d6d756c74692d637572736f722d706c75732f6769662f73686f77636173652e676966](https://cloud.githubusercontent.com/assets/19866231/23032826/ff22e12c-f45c-11e6-8b41-0263fd185dbb.gif)

**Comment by alexandrudima** [comment]
This could be done from an extension. Out of the box, we ship with ctrl+alt+down that adds a cursor down for each cursor or alt+click. 

In this specific example, I would select `': '` and use `ctrl+d` to swiftly and efficiently select all those instances.

**Comment by cladelpino** [comment]
I find myself often in a greater need for a keyboard shortcut to disable all cursors and return to the "original" one.

**Comment by eromoe** [comment]
In sublime `Ctrl-Alt-Down/Up`,  create cursors,  `ctrl+ x`  would  cut all cursors lines.
But vscode not, only the original cursor line.

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

@mjbvz , sorry, but simply remove and re-add the label "* extension-candidate" to the bot to close the question? Really? Being an extension-candidate does not justify closing ... if someone is interested it will be easier to find.

Yes, in fact this issue was based on an Atom extension, but that might well be native to VSCode. I urge you to reopen the issue, or show a justification for closing it other than "Many Houses Open" or "Few Votes." For a significant vote, not to the point of being a priority.

**Comment by Tekbr** [comment]
@mjbvz  or @alexandrudima , did you reopen the issue?

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

A lot, but thank you very much @alexandrudima , for reopening the issue !! ❤️ 😍 

I'm happy for your understanding and other users still want this feature (either natively or by extension) 😄 👍

**Comment by vjdw** [comment]
I don't know how this compares to how it's done in Atom, but the [cursor-tools](https://marketplace.visualstudio.com/items?itemName=naumovs.cursor-tools) extension solves this issue for me.

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

@vjdw The Atom extension works most intuitively to date for me.
Try it.
It has more features. See the gif in the first post.

**Comment by eromoe** [comment]
@vjdw 
Sorry, this ext doesn't work at all.  `ctrl+alt + downarrow` would not add a cursor in down position of current.

**Comment by ricardofreeman** [comment]
[cursor-tools](https://marketplace.visualstudio.com/items?itemName=naumovs.cursor-tools) extension works @eromoe, check your keybindings. You might have them already being applied to something else.
I set the option to mark the cursor with Alt+X and the option to enable all marks with Alt+Z and it works like a charm:
![cursor_tools](https://user-images.githubusercontent.com/17456791/41705778-f02b662e-7531-11e8-9f83-d00bf5531e1a.gif)

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

@ricardofreeman It still does not convince me ... Maybe you can explain how it worked! 🤔 

1- After activating some cursors, edit what you need, how can I disable only one cursor by navigating the file with (up, down, left, right) without affecting the position of the other cursors?
2 - After activating some cursors, edit what you need, how can I activate another cursor by navigating the file with (up, down, left, right) without affecting the position of the other cursors?

**Comment by ricardofreeman** [comment]
@Tekbr you're right, I wasn't able to change the active cursors once enabled. Perhaps the developer @sergiirocks will allow this in the future?
Still, it's better than nothing :)

**Comment by tonix-tuft** [comment]
Cursor tools FTW.

**Comment by idanpa** [comment]
Solved similar problem using different approach - [extension for context aware multi-cursor](https://marketplace.visualstudio.com/items?itemName=idanp.smartmulticursor).

**Comment by mitchellevan** [comment]
Please add the "accessibility" label to this issue. It's important for hands-free voice control or any other assistive technology that emulates a keyboard. Recently added [screen reader support for multiple cursors](https://code.visualstudio.com/updates/v1_55#_accessibility) further increases the importance of full keyboard access.

**Comment by Nabav** [comment]
It is relatively a long time that I can't use the [Cursor Tools](https://marketplace.visualstudio.com/items?itemName=naumovs.cursor-tools). Any idea?

**Comment by KingQuarter** [comment]
I can confirm that [Cursor tools](https://marketplace.visualstudio.com/items?itemName=naumovs.cursor-tools), along with its fork, [Cursor tools by ctf0](https://marketplace.visualstudio.com/items?itemName=ctf0.cursor-tools), doesn't work as of VS Code v1.70.2. There is another extension directly inspired by it, [Cursor Drop](https://atom.io/packages/cursor-drop); however, that's for Atom.

I've searched to the ends of the earth in the [Marketplace](https://marketplace.visualstudio.com/vscode), manually combing through each result for the query in "cursor", "multi-cursor", "multicursor", "anchor", "marker", "caret", and "indicator". I've probably looked at at least a thousand extensions, but I haven't found one that can do what Cursor tools could.

I've concluded that no such working extension for the latest release of VS Code exists at the moment. The next best thing to do would be to fork Cursor tools and fix it for the latest release and keep upvoting this issue.

**Comment by Nabav** [comment]
Need this from deepest point of my heart. :sob: :broken_heart:

**Comment by srares13** [comment]
Try this, maybe its an answer: https://marketplace.visualstudio.com/items?itemName=srares13.kcs
Inform me of any issues you have. If you want, for that you can use the Issues tab of the extension's repository.

**Comment by Nabav** [comment]
> Try this, maybe its an answer: https://marketplace.visualstudio.com/items?itemName=srares13.kcs Inform me of any issues you have. If you want, for that you can use the Issues tab of the extension's repository.

What a good news. I should celebrate it today. :wine_glass: 

The old extension was using `Alt` + `X` and `Alt` + `Y` instead of `PgDn` a `PgUp` keys as default and it was fantastic.

Thanks a lot for this gift @srares13 .

**Comment by zachpoblete** [comment]
@srares13 you are an absolute legend. Thank you for taking the initiative.

@Nabav Actually, the old extension used <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Down</kbd> and <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Up</kbd>. I think you are misremembering because in your [review of the original Cursor tools](https://marketplace.visualstudio.com/items?itemName=naumovs.cursor-tools&ssr=false#review-details) (all the way back in 2018!), you said you rebinded the shortcuts to <kbd>Alt</kbd>+<kbd>X</kbd> and <kbd>Alt</kbd>+<kbd>Y</kbd>.

**Comment by Nabav** [comment]
> @srares13 you are an absolute legend. Thank you for taking the initiative.
> 
> @Nabav Actually, the old extension used Ctrl+Alt+Down and Ctrl+Alt+Up. I think you are misremembering because in your [review of the original Cursor tools](https://marketplace.visualstudio.com/items?itemName=naumovs.cursor-tools&ssr=false#review-details) (all the way back in 2018!), you said you rebinded the shortcuts to Alt+X and Alt+Y.

Yes, you are right. Sorry for that. 🙈

**Comment by Malix-off** [comment]
At first I thought it was doable with `createCursor` but apparently not.
Do someone know what the `createCursor` keyboard shortcut setting do?

**Comment by srares13** [comment]
I come back with a comment just to formally respond to something that was requested as well:

> 1- After activating some cursors, edit what you need, how can I disable only one cursor by navigating the file with (up, down, left, right) without affecting the position of the other cursors?
2 - After activating some cursors, edit what you need, how can I activate another cursor by navigating the file with (up, down, left, right) without affecting the position of the other cursors?

You can achieve both of those points with the [Keyboard Cursors and Selections](https://marketplace.visualstudio.com/items?itemName=srares13.kcs) extension. This is made possible by leveraging features of the extension like:

_If you place an inactive cursor or selection over another, the existing one will be removed_
_Undo-Redo system_
_Inactive cursors and selections react to document edits_

These features are all described within the [home page](https://marketplace.visualstudio.com/items?itemName=srares13.kcs) of the extension.
